### DISCUSSION TITLE
Including data files from external modules

### DISCUSSION THREAD

**Discussion by ben-briant** [pyinstaller/pyinstaller#7400] [Help] [answered]
Hello! I'm looking for the best way to include data files _from an external module_ in my bundled executable.

I've seen this page https://pyinstaller.org/en/stable/spec-files.html#using-data-files-from-a-module which seems to discuss including data files from modules within my own application, but not including data files from installed modules.

For context, I've got an external library which relies on `.txt` files packaged in that library, and pyinstaller isn't including those in the bundled executable, causing the library to fail while running.

Really appreciate any help/advice you can give!

**Comment by rokm** [comment, marked as answer]
[`--collect-data`](https://pyinstaller.org/en/stable/usage.html#cmdoption-collect-data) on command line or [`PyInstaller.utils.hooks.collect_data_files`](https://pyinstaller.org/en/stable/hooks.html#PyInstaller.utils.hooks.collect_data_files) in a custom hook.

**Reply by ben-briant** [reply]
Ah -- specifically, the files are in `venv/lib/python3.6/site-packages/foo/bar.txt`. At the moment, I've specified that path in the `datas` section of the `.spec` file and everything works fine (thanks!), but was hoping there might be a more robust way to specify data files coming from a package?

**Reply by rokm** [reply]
Either you write a hook for that package that calls `collect_data_files`, or you add `collect_data_files` call to the spec file and pass the resulting data to `Analysis` via `datas` argument (which is what happens when you run `pyinstaller --collect-data foo myscript.py` - PyInstaller adds a `collect_data_files('foo')` to the top of the generated spec file).

**Reply by ben-briant** [reply]
Ah, that's worked beautifully. Thank you so much!
